substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ResourceConfigurationParser.java

```diff
@@ -60,9 +60,35 @@ private void parseTopLevelObject(Map<String, Object> obj) {
             }
         }
         if (resourcesObject != null) {
-            List<Object> resources = asList(resourcesObject, "Attribute 'resources' must be a list of resources");
-            for (Object object : resources) {
-                parseEntry(object, "pattern", registry::addResources, "resource descriptor object", "'resources' list");
+            if (resourcesObject instanceof Map) { // New format
+                Object includesObject = null;
+                Object excludesObject = null;
+                for (Map.Entry<String, Object> pair : ((Map<String, Object>) resourcesObject).entrySet()) {
```

Comment:
If you use `(((Map<String, Object>) resourcesObject).entrySet()).forEach((key, value) -> {` you avoid the `Map.Entry<String, Object>`-type orgy.

Reply:
Nah doesn't work because
```
    Object includesObject = null;
    Object excludesObject = null;
```
cannot be assigned to from the lambda.

Reply:
How about
```patch
--- substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ResourceConfigurationParser.java	(revision d2a6f30c05cb24f802e1d828f2eb03f7124dab8e)
+++ substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ResourceConfigurationParser.java	(date 1602691803931)
@@ -61,17 +61,9 @@
         }
         if (resourcesObject != null) {
             if (resourcesObject instanceof Map) { // New format
-                Object includesObject = null;
-                Object excludesObject = null;
-                for (Map.Entry<String, Object> pair : ((Map<String, Object>) resourcesObject).entrySet()) {
-                    if ("includes".equals(pair.getKey())) {
-                        includesObject = pair.getValue();
-                    } else if ("excludes".equals(pair.getKey())) {
-                        excludesObject = pair.getValue();
-                    } else {
-                        throw new JSONParserException("Unknown attribute '" + pair.getKey() + "' (supported attributes: name) in resource definition");
-                    }
-                }
+                Map<String, Object> map = (Map<String, Object>) resourcesObject;
+                Object includesObject = map.get("includes");
+                Object excludesObject = map.get("excludes");
 
                 List<Object> includes = asList(includesObject, "Attribute 'includes' must be a list of resources");
                 for (Object object : includes) {
```

Reply:
Yes, it's definitely neater but in this case, we do not check for extra unknown attributes anymore, like it is done just above:

```
for (Map.Entry<String, Object> pair : obj.entrySet()) {
            if ("resources".equals(pair.getKey())) {
                resourcesObject = pair.getValue();
            } else if ("bundles".equals(pair.getKey())) {
                bundlesObject = pair.getValue();
            } else {
                throw new JSONParserException("Unknown attribute '" + pair.getKey() + "' (supported attributes: name) in resource definition");
            }
        }
```

What do you think? I can also change both parts with your solution...

Reply:
> Yes, it's definitely neater but in this case, we do not check for extra unknown attributes anymore

Rrrrright. We do not want to break this strict checking behaviour for saving a few lines of code.
Sorry for the noise.